src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -646,6 +646,51 @@ type internal TypeCheckInfo
         let items = items |> RemoveExplicitlySuppressed g
         items, nenv.DisplayEnv, m
 
+    /// Find union cases and compatible active patterns when the item under the cursor position is an identifier with the type of a DU.
+    let GetUnionCasesAndActivePatternsEnvironmentLookupResolutionsAtPosition cursorPos =
+        let rec doesActivePatternTakeTypeAsInput g ty paramType =
```

Comment:
Could you take this out as a separate function called `ActivePatternFeasiblyAcceptsTypeAsInput`.  Perhaps put it in `TypeRelations.fs`.  The `Feasibly` indicates that the thing is approximate - e.g. any generic variable will be considered to accept the type, even if its constraints are not met.  That's ok.